tests/ui/test_ui_tools.py

```diff
@@ -568,64 +568,34 @@ def test_mouse_event(self, mocker, stream_view, widget_size):
+
         stream_view.keypress(size, key)
+        assert stream_view.focus_index_before_search == 3
         stream_view.set_focus.assert_called_once_with("header")
 
     @pytest.mark.parametrize("key", keys_for_command("GO_BACK"))
     def test_keypress_GO_BACK(self, mocker, stream_view, key, widget_size):
         size = widget_size(stream_view)
         mocker.patch.object(stream_view, "set_focus")
         mocker.patch.object(stream_view.stream_search_box, "reset_search_text")
-        stream_view.keypress(size, key)
-        stream_view.set_focus.assert_called_once_with("body")
-        assert stream_view.stream_search_box.reset_search_text.called
-        assert stream_view.log == self.streams_btn_list
-
-    @pytest.mark.parametrize("search_streams_key", keys_for_command("SEARCH_STREAMS"))
-    @pytest.mark.parametrize("go_back_key", keys_for_command("GO_BACK"))
-    @pytest.mark.parametrize(
-        "current_focus, stream",
-        [
-            (0, "FOO"),
-            (2, "fan"),
-            (4, "BOO"),
-        ],
-    )
-    def test_return_to_focus_after_search(
-        self,
-        mocker,
-        stream_view,
-        widget_size,
-        current_focus,
-        stream,
-        search_streams_key,
-        go_back_key,
-    ):
-        # Initialize log
-        stream_view.streams_btn_list = [
-            mocker.Mock(stream_name=stream_name)
-            for stream_name in ["FOO", "foo", "fan", "boo", "BOO"]
-        ]
-        stream_view.log.extend(stream_view.streams_btn_list)
+        stream_view.streams_btn_list = ["FOO", "foo", "fan", "boo", "BOO"]
+        stream_view.focus_index_before_search = 3
 
-        # Set initial stream focus to 'current_focus' and name to 'stream'
-        stream_view.log.set_focus(current_focus)
-        stream_view.focus_index_before_search = current_focus
-        previous_focus = stream_view.log.get_focus()[1]
-        previous_focus_stream_name = stream
+        # During search
+        stream_view.log.clear()
+        stream_view.log.extend(stream_view.streams_btn_list[3])
+        stream_view.log.set_focus(1)
 
-        # Toggle Stream Search
-        size = widget_size(stream_view)
-        stream_view.keypress(size, search_streams_key)
-
-        # Exit Stream Search
-        stream_view.keypress(size, go_back_key)
+        # Exit search
+        stream_view.keypress(size, key)
 
-        # Obtain new stream focus
-        new_focus = stream_view.log.get_focus()[1]
-        new_focus_stream_name = stream_view.log[new_focus].stream_name
-        assert new_focus == previous_focus
-        assert previous_focus_stream_name == new_focus_stream_name
+        # Check state reset after search
+        stream_view.set_focus.assert_called_once_with("body")
+        assert stream_view.stream_search_box.reset_search_text.called
+        assert stream_view.log == stream_view.streams_btn_list
+        assert stream_view.log.get_focus()[1] == stream_view.focus_index_before_search
```

Comment:
Let's also check whether `set_focus` was called as well?

Reply:
Isn't it same as checking whether focus_index was updated? Doing either tests the same functionality?

Reply:
Right, it's a bit redundant here.